src/Core/Services/Implementations/AppleIapService.cs

```diff
@@ -85,15 +85,16 @@ public async Task SaveReceiptAsync(AppleReceiptStatus receiptStatus, Guid userId
                 receipt.ContainsKey("UserId") ? new Guid(receipt["UserId"]) : (Guid?)null);
         }
 
-        private async Task<AppleReceiptStatus> GetReceiptStatusAsync(string receiptData, bool prod = true,
+        // Internal for testing
+        internal async Task<AppleReceiptStatus> GetReceiptStatusAsync(string receiptData, bool prod = true,
             int attempt = 0, AppleReceiptStatus lastReceiptStatus = null)
         {
             try
             {
                 if (attempt > 4)
                 {
-                    throw new Exception("Failed verifying Apple IAP after too many attempts. Last attempt status: " +
-                        lastReceiptStatus?.Status ?? "null");
```

Comment:
This could have also been handled with parens (the O.G. of order-of-operations directives)
```c#
(lastReceiptStatus?.Status ?? "null")
```